### PR TITLE
Bulk ingestion of test data

### PR DESCRIPTION
**Pull request by adwk67**
- generate bulk-loaded data
- add a unit test
- add documentation and an example to run against an HBase cluster
- Simple test on a single node kind cluster:
```
bash-4.4$ java -cp '/tmp/hbase-stackable/local/jars/hbase-gis-1.0.0.jar:/stackable/conf' tech.stackable.gis.hbase.BulkIngest wifi a 100000 -75.99,-75.01,44.01,44.99
...
Geohashed 100000 records in 17704ms.
bash-4.4$ java -cp '/tmp/hbase-stackable/local/jars/hbase-gis-1.0.0.jar:/stackable/conf' \
>     tech.stackable.gis.hbase.WithinQuery \
>     remote \
>     "POLYGON ((-76.0 44.97, \
>         -76.0 45.0, \
>         -75.9 45.0, \
>         -75.9 44.97, \
>         -76.0 44.97))"
...
Query matched 195 points in 968ms.
bash-4.4$ 
```

Short test on Openshift:
- 1m rows loaded to 6 region servers (4 regions created)
- filter query as above: finds 1870 rows in 2.5 secs (zero optimization i.e. rowkey is ignored).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +61,19 @@ Load data
...
+    java -cp '/tmp/hbase-stackable/local/jars/hbase-gis-1.0.0.jar:/stackable/conf' \
+    tech.stackable.gis.hbase.WithinQuery \
+    remote \
+    "POLYGON ((-76.0 44.97, \
+        -76.0 45.0, \
+        -75.9 45.0, \
+        -75.9 44.97, \
+        -76.0 44.97))"
+
+You should a subset of the bulk-loaded back as the results of this query (due to the way that the query polygon and the limits of the randomly generated data have been defined.)
```
Comment (razvan) [inline README.md]:
```suggestion
You should get a subset of the bulk-loaded back as the results of this query (due to the way that the query polygon and the limits of the randomly generated data have been defined.)
```

File: src/main/java/tech/stackable/gis/hbase/BulkIngest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        String[] limits = String.valueOf(args[3]).split(",");
+
+        // we need all 4
+        if (limits.length != 4) {
+            System.out.println(usage);
+            System.exit(0);
+        }
+
+        // generate random co-ordinates within a fixed region
+        PrimitiveIterator.OfDouble lon_iter = new Random().doubles(Double.valueOf(limits[0]), Double.valueOf(limits[1])).iterator();
```
Comment (razvan) [inline src/main/java/tech/stackable/gis/hbase/BulkIngest.java]: maybe a small sanity check that limits[0] < limits[1] and so on

File: src/test/java/TestWithinFilter.java [hunk 3]
```diff
@@ -204,12 +212,50 @@ public void testLinePointsWithRectangleFilter() throws Exception {
...
-        withinFilter = new WithinFilter(query, "wifi".getBytes(), "a".getBytes(), "lat".getBytes(), "lon".getBytes());
+        withinFilter = new WithinFilter(query, "wifi".getBytes(), FAMILY_B, "lat".getBytes(), "lon".getBytes());
         filters = new FilterList(withinFilter);
-        results = queryWithFilterAndRegionScanner(REGION, filters, FAMILY, COLUMNS_RECTANGLE_CHECK);
+        results = queryWithFilterAndRegionScanner(REGION, filters, FAMILY_B, COLUMNS_RECTANGLE_CHECK);
         assertEquals(3, results);
     }
 
+    @Test
+    public void testBulkIngest()  throws Exception {
```
Comment (razvan) [inline src/test/java/TestWithinFilter.java]: I would have put this in a separate test file

File: src/test/java/TestWithinFilter.java [hunk 4]
```diff
@@ -204,12 +212,50 @@ public void testLinePointsWithRectangleFilter() throws Exception {
...
+        int row_count = 1000;
+        long start = System.currentTimeMillis();
+
+        // generate random co-ordinates within a fixed region
+        PrimitiveIterator.OfDouble lon_iter = new Random().doubles(-75.99, -75.01).iterator();
+        PrimitiveIterator.OfDouble lat_iter = new Random().doubles(44.01, 44.99).iterator();
```
Comment (razvan) [inline src/test/java/TestWithinFilter.java]: i see what you did here :)

File: src/main/java/tech/stackable/gis/hbase/BulkIngest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+            admin.flush(tableName);
+        } finally {
+            table.close();
+            connection.close();
+            admin.close();
+        }
+
+        long end = System.currentTimeMillis();
+        System.out.printf("Geohashed %s records in %sms.%n", row_count, end - start);
```
Comment (razvan) [inline src/main/java/tech/stackable/gis/hbase/BulkIngest.java:82]: an idea: you could  put the block starting from line 53 to 71 in a separate function and just call that in test.

**Review by razvan — APPROVED** [review]
nice

File: src/main/java/tech/stackable/gis/hbase/BulkIngest.java (hunk 5 above)
Comment (adwk67) [reply src/main/java/tech/stackable/gis/hbase/BulkIngest.java:82]: If I do that I have the `Table` object in there which means I need a cluster. Retrieving just the `Put` means I can use it with a `Region` in isolation.

File: src/main/java/tech/stackable/gis/hbase/BulkIngest.java (hunk 2 above)
Comment (adwk67) [reply src/main/java/tech/stackable/gis/hbase/BulkIngest.java]: Done.

File: src/test/java/TestWithinFilter.java (hunk 3 above)
Comment (adwk67) [reply src/test/java/TestWithinFilter.java]: Done. Common functionality moved to abstract class.

**Pull request merged**